html/browsers/browsing-the-web/navigating-across-documents/initial-empty-document/resources/helpers.js

```diff
@@ -0,0 +1,99 @@
+const url_204 = "/common/blank.html?pipe=status(204)";
+
+window.waitForLoad = (t, iframe, urlRelativeToThisDocument) => {
+  return new Promise(resolve => {
+    iframe.addEventListener("load", t.step_func(() => {
+      assert_equals(iframe.contentWindow.location.href, (new URL(urlRelativeToThisDocument, location.href)).href);
+
+      // Wait a bit longer to ensure all history stuff has settled, e.g. the document is "completely loaded"
+      // (which happens from a queued task).
+      setTimeout(resolve, 0);
+    }), { once: true });
+  });
+};
+
+window.waitForMessage = (t, message) => {
+  return new Promise(resolve => {
+    window.addEventListener("message", t.step_func((event) => {
+      if (event.data == message)
+        resolve();
+    }), { once: true });
```

Comment:
You could try removing `{ once: true }` here to see if that helps the tests pass, e.g. if multiple messages are being sent somehow. If it doesn't help though then keeping it is best.

Reply:
Thanks, I think this helped some tests